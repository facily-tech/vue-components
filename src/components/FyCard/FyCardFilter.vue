<template>
  <fy-card class="fy-card-filter pa-2 pa-md-4 pb-md-0" v-bind="propsAndAttrs" v-on="$listeners">
    <v-form ref="form">
      <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
    </v-form>
  </fy-card>
</template>

<script lang="ts">
import Vue from 'vue';
import FyCard from './FyCard.vue';

export default Vue.extend({
  name: 'FyCardFilter',

  components: {
    FyCard,
  },

  computed: {
    propsAndAttrs(): Record<string, unknown> {
      return {
        ...this.$attrs,
        ...this.$props,
      };
    },
  },
});
</script>
