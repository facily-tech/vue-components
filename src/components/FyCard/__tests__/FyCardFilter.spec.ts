import Vue from 'vue';
import Router from 'vue-router';
import FyCardFilter from '../FyCardFilter.vue';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('FyCardFilter', () => {
  // eslint-disable-next-line
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<any>;
  let router: Router;
  let localVue: typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);

    mountFunction = (options = {}) => {
      return mount(FyCardFilter, {
        localVue,
        vuetify: new Vuetify(),
        router,
        ...options,
      });
    };
  });

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });
});
