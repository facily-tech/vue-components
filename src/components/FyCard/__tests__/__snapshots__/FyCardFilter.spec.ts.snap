// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyCardFilter should render component and match snapshot 1`] = `
<div class="fy-card-filter pa-2 pa-md-4 pb-md-0 v-card v-sheet theme--light elevation-0">
  <form novalidate="novalidate" class="v-form"></form>
</div>
`;
